### PR TITLE
Permission user

### PR DESCRIPTION
**Pull request by phucynwa**

### CODE REVIEW

**Comment by phucynwa** [comment]
https://dev.framgia.com/issues/184326

**Comment by phucynwa** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -3,6 +3,11 @@ class UsersController < ApplicationController
   before_action :logged_in_user, except: %i(new create)
   before_action :not_logged_in, only: %i(new create)
   before_action :correct_user, only: %i(edit update)
+  before_action :verify_staff, only: :index
```
[the remark this answers is not in the document]
Comment (phucynwa) [reply app/controllers/users_controller.rb]: @mdmahmudurrahman no, admin is also a staff

**Comment by phucynwa** [comment]
ready

**Comment by phucynwa** [comment]
ready

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -26,7 +26,11 @@ def not_logged_in
     end
   end
 
-  def vetify_admin
+  def verify_staff
+    redirect_to root_path unless current_user&.staff? || current_user&.admin?
```
Comment (huuphuc93) [inline app/controllers/application_controller.rb]: add flash if use cannot access same with verify_admin

**Comment by phucynwa** [comment]
ready

**Comment by phucynwa** [comment]
ready

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -26,7 +26,17 @@ def not_logged_in
     end
   end
 
-  def vetify_admin
-    redirect_to root_path unless current_user&.admin?
+  def verify_staff
+    unless current_user&.staff? || current_user&.admin?
+      flash[:warning] = t "users.invalid_action"
+      redirect_to root_path
+    end
```
Comment (mdmahmudurrahman) [inline app/controllers/application_controller.rb]: Can you re-write using the following
```ruby
return if current_user&.staff? || current_user&.admin?
flash[:warning] = t "users.invalid_action"
redirect_to root_path
```

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -26,7 +26,17 @@ def not_logged_in
...
+  end
+
+  def verify_admin
+    unless current_user&.admin?
+      flash[:warning] = t "users.invalid_action"
+      redirect_to root_path
+    end
```
Comment (mdmahmudurrahman) [inline app/controllers/application_controller.rb]:
```ruby
return if current_user&.admin?
flash[:warning] = t "users.invalid_action"
redirect_to root_path
```

File: app/views/users/_user.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<tr class="user-row">
+  <td><%= user.id %></td>
+  <td>
+    <%= link_to user, class: "user-name" do %>
+      <%= gravatar_for user, size: Settings.gravatar_size %>
+      <%= user.name %>
+    <% end %>
+    <% if user.admin? %>
+      <span class="label label-danger">Admin</span>
```
Comment (mdmahmudurrahman) [inline app/views/users/_user.html.erb]: I18n

File: app/views/users/_user.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    <% elsif user.staff? %>
+      <span class="label label-primary">Staff</span>
```
Comment (mdmahmudurrahman) [inline app/views/users/_user.html.erb]: Same here

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -1,14 +1,64 @@
...
-        <%= @user.name %>
-      </h1>
-      <%= @user.email %>
-      <br>
-      <%= @user.phone %>
-    </section>
-  </aside>
-</div>
+<section class="user-home-wrapper">
+
```
Comment (mdmahmudurrahman) [inline app/views/users/show.html.erb]: Can I request you to remove this unnecessary empty line

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by phucynwa** [comment]
ready

File: app/controllers/application_controller.rb [hunk 8]
```diff
@@ -26,7 +26,17 @@ def not_logged_in
     end
   end
 
-  def vetify_admin
-    redirect_to root_path unless current_user&.admin?
+  def verify_staff
+    return if current_user&.staff? || current_user&.admin?
+      flash[:warning] = t "users.invalid_action"
+      redirect_to root_path
+    end
```
Comment (mdmahmudurrahman) [inline app/controllers/application_controller.rb]: Em, remove this redundant end 
```ruby
def verify_staff
  return if current_user&.staff? || current_user&.admin?
  flash[:warning] = t "users.invalid_action"
  redirect_to root_path
end 
```

File: app/controllers/application_controller.rb [hunk 9]
```diff
@@ -26,7 +26,17 @@ def not_logged_in
...
+  end
+
+  def verify_admin
+    return if current_user&.admin?
+      flash[:warning] = t "users.invalid_action"
+      redirect_to root_path
+    end
```
Comment (mdmahmudurrahman) [inline app/controllers/application_controller.rb]: Same here

**Comment by mdmahmudurrahman** [comment]
commented

File: app/controllers/application_controller.rb (hunk 8 above)
Comment (phucynwa) [reply app/controllers/application_controller.rb]: @mdmahmudurrahman i'm so sorry :(

**Comment by mdmahmudurrahman** [comment]
(merged)
CC: [To:2076564] Nguyen Huy Hung C (93)

**Pull request merged**